core/src/test/scala/pureconfig/PureconfSuite.scala

```diff
@@ -227,6 +225,51 @@ class PureconfSuite extends FlatSpec with Matchers with OptionValues with TryVal
     }
   }
 
+  sealed trait AnimalConfig
+  case class DogConfig(age: Int) extends AnimalConfig
+  case class CatConfig(age: Int) extends AnimalConfig
+  case class BirdConfig(canFly: Boolean) extends AnimalConfig
+
+  it should s"be able to save and load ${classOf[AnimalConfig]}" in {
+    List(DogConfig(12), CatConfig(3), BirdConfig(true)).foreach { expectedConfig =>
+      saveAndLoadIsIdentity[AnimalConfig](expectedConfig)
+    }
+  }
+
+  it should s"read and write disambiguation information on sealed families by default" in {
+    withTempFile { configFile =>
+      val conf = ConfigFactory.parseString("{ type = dogconfig, age = 2 }")
+      loadConfig[AnimalConfig](conf) should be(Success(DogConfig(2)))
+
+      saveConfigAsPropertyFile[AnimalConfig](DogConfig(2), configFile, overrideOutputPath = true)
+      loadConfig[TypesafeConfig](configFile).map(_.getString("type")) should be(Success("dogconfig"))
+    }
+  }
+
+  it should s"allow using different strategies for disambiguating between options in a sealed family" in {
+    withTempFile { configFile =>
+      implicit val hint = new FieldCoproductHint[AnimalConfig]("which-animal") {
+        override def fieldValue(name: String) = name.dropRight("Config".length)
+      }
+
+      val conf = ConfigFactory.parseString("{ which-animal = Dog, age = 2 }")
+      loadConfig[AnimalConfig](conf) should be(Success(DogConfig(2)))
+
+      saveConfigAsPropertyFile[AnimalConfig](DogConfig(2), configFile, overrideOutputPath = true)
+      loadConfig[TypesafeConfig](configFile).map(_.getString("which-animal")) should be(Success("Dog"))
+    }
+
+    withTempFile { configFile =>
+      implicit val hint = new FirstSuccessCoproductHint[AnimalConfig]
+
+      val conf = ConfigFactory.parseString("{ canFly = true }")
+      loadConfig[AnimalConfig](conf) should be(Success(BirdConfig(true)))
+
+      saveConfigAsPropertyFile[AnimalConfig](DogConfig(2), configFile, overrideOutputPath = true)
+      loadConfig[TypesafeConfig](configFile).map(_.hasPath("type")) should be(Success(false))
+    }
+  }
+
```

Comment:
Can you also add some tests where the implicit coproduct derivation fails (namely when `CollidingKeysException` is returned and the field of `FieldCoproductHint` is missing)?